superset/utils/core.py

```diff
@@ -1065,6 +1065,9 @@ def to_adhoc(
     elif expression_type == "SQL":
         result.update({"sqlExpression": filt.get(clause)})
 
+    deterministic_name = md5_sha_from_dict(result)
+    result["filterOptionName"] = deterministic_name
```

Comment:
I'm unclear what the impact of changing the `filterOptionName` value is here. @rusackas @villebro

Reply:
Yeah, I agree we should get some 👀 on this from folks more familiar. AFAICT, this preserves the existing behavior of generating a unique key (which I assume is needed on clients) but also keeps it deterministic so that hashing `form_data` for will work properly (both with async/non-async caching).

Reply:
With a deterministic `filterOptionName` we of course now have the risk of having duplicate `filterOptionNames` which probably is precisely what the uuid4 key is aiming to avoid (could cause trouble in React components if these get returned to the frontend). So maybe we should deduplicate filters, too, which in edge cases would avoid unnecessary cache misses (=if the exact same filter has been defined twice vs another query where the same filter is only defined once)?

Reply:
Hmm... can you point to a specific example of where that might be an issue? Wouldn't each filter have different attributes? If not, is it a bug on the client that the same filters for the same chart would be added more than once? The idea is this should be deterministic for the same filter value, but the set of filters should have different properties, causing a different sha.